integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/thrift/ExecuteHQLStatement.java

```diff
@@ -102,27 +103,40 @@ public ExecuteHQLStatement() {
     }
 
     @Override
-    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
-        final ComponentLog logger = getLog();
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws RuntimeException {
         FlowFile flowFile = session.get();
         if (flowFile == null) {
             return;
         }
 
-        final ThriftService thriftService = context.getProperty(THRIFT_SERVICE).asControllerService(ThriftService.class);
         final String ddlQuery = context.getProperty(SQL_DDL_STATEMENT).evaluateAttributeExpressions(flowFile).getValue();
+        String[] hiveStatements = StringUtils.split(ddlQuery, ';');
+        final ThriftService thriftService = context.getProperty(THRIFT_SERVICE).asControllerService(ThriftService.class);
+
+        executeStatements(session, flowFile, hiveStatements, thriftService);
+
+
+    }
+
+    public void executeStatements(ProcessSession session, FlowFile flowFile, String[] hiveStatements, ThriftService thriftService) {
+        final ComponentLog logger = getLog();
 
         final StopWatch stopWatch = new StopWatch(true);
 
         try (final Connection con = thriftService.getConnection();
              final Statement st = con.createStatement()) {
+            boolean result = false;
+
+            for (String statement : hiveStatements) {
+                result = st.execute(statement);
+            }
 
-            boolean result = st.execute(ddlQuery);
             session.getProvenanceReporter().modifyContent(flowFile, "Execution result " + result, stopWatch.getElapsed(TimeUnit.MILLISECONDS));
             session.transfer(flowFile, REL_SUCCESS);
         } catch (final Exception e) {
-            logger.error("Unable to execute SQL DDL {} for {} due to {}; routing to failure", new Object[]{ddlQuery, flowFile, e});
+            logger.error("Unable to execute SQL DDL");
```

Comment:
I'm concerned removing the DDL leads to a lack of debugging capability.  Can you add "statements, flowfile, e" to the log message?